src/App.tsx

```diff
@@ -1,14 +1,51 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api/todos';
+import { filterByOption, filterByQuery } from './services/filter';
+import { DispatchContext, StateContext } from './Store';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const dispatch = useContext(DispatchContext);
+  const {
+    todos,
+    openedTodo,
+    filter,
+    isModalOpened,
+    isLoadingTodos,
+  } = useContext(StateContext);
+
+  const filterTodos = useCallback((todosToFilter: Todo[]) => {
+    const { option, query } = filter;
+
+    return filterByQuery(filterByOption(todosToFilter, option), query);
+  }, [filter]);
+
+  const [filteredTodos, setFilteredTodos] = useState(filterTodos(todos));
```

Comment:
You shouldn't recreate the initial state
https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state